cpp/src/transform/nans_to_nulls.cu

```diff
@@ -37,17 +37,28 @@ struct dispatch_nan_to_null {
         auto input_device_view_ptr = column_device_view::create(input, stream);
         auto input_device_view = *input_device_view_ptr;
 
-        auto pred = [input_device_view] __device__ (cudf::size_type idx) {
-            return not (std::isnan(input_device_view.element<T>(idx)) || input_device_view.is_null(idx)); 
-        };
+        if (input.nullable()) {
+            auto pred = [input_device_view] __device__ (cudf::size_type idx) {
+                return not (std::isnan(input_device_view.element<T>(idx)) || input_device_view.is_null(idx));
```

Comment:
You can use the `is_null_no_check` version now that it's guaranteed that the input has a mask.